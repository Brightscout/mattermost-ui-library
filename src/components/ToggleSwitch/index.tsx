import React from 'react';

// Constants
import {ToggleSwitchLabelPositioningValues} from '@Constants';

type ToggleSwitchProps = {
    active: boolean;
    onChange: (active: boolean) => void;
    label?: string;
    labelPositioning?: ToggleSwitchLabelPositioning;
}

const ToggleSwitch = ({
    active,
    onChange,
    label,
    labelPositioning = ToggleSwitchLabelPositioningValues.Left,
}: ToggleSwitchProps): JSX.Element => (
    <div className={`toggle-switch-container margin-bottom-20 padding-bottom-10 d-flex align-items-center ${labelPositioning === ToggleSwitchLabelPositioningValues.Right && 'flex-row-reverse'}`}>
        {label && <span className={labelPositioning === ToggleSwitchLabelPositioningValues.Left ? 'toggle-switch-label--left margin-right-10' : 'toggle-switch-label--right margin-left-10'}>{label}</span>}
        <label className='toggle-switch position-relative d-inline-block margin-0 cursor-pointer'>
            <input
                type='checkbox'
                className='toggle-switch__input'
                checked={active}
                onChange={(e) => onChange(e.target.checked)}
            />
            <span className='toggle-switch__slider position-absolute cursor-pointer'/>
        </label>
    </div>
);

export default ToggleSwitch;
