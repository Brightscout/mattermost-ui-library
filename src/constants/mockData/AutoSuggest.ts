// Mock data for "AutoSuggest" component tests
export const onInputValueChange = jest.fn();
export const onOptionClick = jest.fn();
export const suggestionConfig = {
    suggestions: [{
        label: 'Label 1',
        value: 'Value 1',
    }, {
        label: 'Label 2',
        value: 'Value 2',
    }, {
        label: 'Label 3',
        value: 'Value 3',
    }],
    renderValue: (suggestion: Record<string, string>) => suggestion.label,
};
