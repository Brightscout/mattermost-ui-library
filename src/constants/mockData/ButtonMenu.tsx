import React from 'react';

// Mock data for "ButtonMenu" component tests
export const Icon = <i className='icon icon-globe' />;
export const onBtnClick = jest.fn();
export const buttons = [{
    text: 'Button Text',
    icon: Icon,
    onClick: onBtnClick,
}, {
    text: 'Button Text 2',
    icon: Icon,
    onClick: onBtnClick,
}];
