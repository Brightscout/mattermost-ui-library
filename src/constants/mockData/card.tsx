import React from 'react';

// Mock data for "card" related component tests

// Mock data for "BaseCard"
export const BaseCardChildren = <div>{'Base card children'}</div>;

// Mock data for "SubscriptionCard"
export const editHandler = jest.fn();
export const deleteHandler = jest.fn();
export const subscriptionCardBody = {
    list: ['List Item 1', 'List Item 2', 'List Item 3'],
    labelValuePairs: [
        {label: 'Label 1', value: 'Value 1'},
        {label: 'Label 2', value: 'Value 2'},
        {label: 'Label 3', value: 'Value 3'},
        {label: 'Label 4', value: 'Value 4'},
    ],
};
// Mock Icons
export const EditIcon = <i className='fa fa-edit' />;
export const DeleteIcon = <i className='fa fa-delete' />;
// Mock channel data
export const channelData = {
    create_at: 19304124124,
    creator_id: 'klsioer239042',
    delete_at: 19304124124,
    display_name: 'Channel Display Name',
    extra_update_at: 19304124124,
    group_constrained: null,
    header: 'Channel Header',
    id: 'jw390r23iofs9032',
    last_post_at: 19304124124,
    last_root_post_at: 19304124124,
    name: 'Channel Name',
    policy_id: null,
    props: null,
    purpose: 'Channel Purpose',
    scheme_id: null,
    shared: null,
    team_display_name: 'Team Display Name',
    team_id: 'ior9823ijoiewro3w',
    team_name: 'Team Name',
    team_update_at: 19304124124,
    total_msg_count: 1,
    total_msg_count_root: 1,
    type: 'P',
    update_at: 19304124124,
};
export const subscriptionCardProps = {
    header: 'Subscription Card Header',
    label: 'Label',
    cardBody: subscriptionCardBody,
    description: 'Description',
    channel: channelData,
    editIcon: EditIcon,
    deleteIcon: DeleteIcon,
    onEdit: editHandler,
    onDelete: deleteHandler,
    className: 'subscription-card-custom-class',
};

// Mock data for "Card" component
export const cardBody = [{
    sectionHeading: 'Section Heading',
    data: [{
        icon: <i className='fa fa-edit' />,
        label: 'Label 1',
    }, {
        icon: <i className='fa fa-link' />,
        label: 'Label 2',
    }],
}];
export const cardProps = {
    cardHeader: 'Card Header',
    cardSubHeader: 'Card sub header',
    cardBody,
    onEdit: editHandler,
    onDelete: deleteHandler,
    editIcon: EditIcon,
    deleteIcon: DeleteIcon,
    className: 'card-custom-class',
};
