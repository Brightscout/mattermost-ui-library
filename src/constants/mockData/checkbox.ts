// Mock data for "Checkbox" component
export const checkboxLabel = 'Checkbox Label';
export const checkboxChangeHandler = jest.fn();
export const changeHandler = jest.fn();
