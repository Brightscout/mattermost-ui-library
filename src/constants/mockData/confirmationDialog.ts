
// Mock data for "ConfirmationDialog" component
export const hideHandler = jest.fn();
export const confirmHandler = jest.fn();
export const confirmationDialogProps = {
    title: 'Dialog Title',
    confirmationMsg: 'Dialog Message',
    show: true,
    onHide: hideHandler,
    loading: true,
    onConfirm: confirmHandler,
    className: 'confirmation-dialog-class',
    error: 'Dialog error',
};
