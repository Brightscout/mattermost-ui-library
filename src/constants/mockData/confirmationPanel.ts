// Mock data for "ConfirmationPanel" component
export const hideHandler = jest.fn();
export const confirmHandler = jest.fn();
export const confirmationPanelProps = {
    confirmationMsg: 'Dialog Message',
    onHide: hideHandler,
    onConfirm: confirmHandler,
    loading: true,
    className: 'confirmation-dialog-class',
    error: 'Dialog error',
};
