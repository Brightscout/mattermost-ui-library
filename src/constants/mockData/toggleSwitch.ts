
// Mock data for "ToggleSwitch" component
export const toggleSwitchOnChangeHandler = jest.fn();
export const toggleSwitchLabel = 'Toggle switch label';
export const toggleSwitchProps = {
    active: true,
    onChange: toggleSwitchOnChangeHandler,
    label: toggleSwitchLabel,
    labelPositioning: 'left' as ToggleSwitchLabelPositioning,
};
