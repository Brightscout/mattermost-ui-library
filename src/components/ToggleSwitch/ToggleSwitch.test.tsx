import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

// Mock data
import {toggleSwitchOnChangeHandler as onChangeHandler, toggleSwitchProps} from '@Constants/mockData/toggleSwitch';

import ToggleSwitch from '.';

describe('ToggleSwitch', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = shallow(<ToggleSwitch {...toggleSwitchProps}/>);
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should render the label correctly', () => {
        expect(component.text().includes(toggleSwitchProps.label)).toBeTruthy();
    });

    it('Should call the onChange when the toggle-switch is clicked', () => {
        component.find('input[type="checkbox"]').simulate('change', {target: {checked: true}});
        expect(onChangeHandler).toHaveBeenCalledTimes(1);
    });

    it('Should keep the label on the left when "labelPositioning" is passed as "left"', () => {
        expect(component.find('.toggle-switch-label--left').exists()).toBeTruthy();
    });
});
