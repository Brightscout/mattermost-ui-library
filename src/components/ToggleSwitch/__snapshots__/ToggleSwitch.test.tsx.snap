// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch Should render correctly 1`] = `
<div
  className="toggle-switch-container margin-bottom-20 padding-bottom-10 d-flex align-items-center false"
>
  <span
    className="toggle-switch-label--left margin-right-10"
  >
    Toggle switch label
  </span>
  <label
    className="toggle-switch position-relative d-inline-block margin-0 cursor-pointer"
  >
    <input
      checked={true}
      className="toggle-switch__input"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="toggle-switch__slider position-absolute cursor-pointer"
    />
  </label>
</div>
`;
